src/planwise/client/utils.cljs

```diff
@@ -89,6 +90,23 @@
   [coll id]
   (remove #(= id (:id %)) coll))
 
+(defn replace-by-id
+  [coll replacement]
+  (let [id (:id replacement)
+        index (->> coll
+                (vec)
+                (keep-indexed #(when (= id (:id %2)) %1))
+                (first))]
+    (assoc (vec coll) index replacement)))
```

Comment:
And probably implement as `(update-by-id coll (:id replacement) constantly replacement)`